go-controller/pkg/node/node.go

```diff
@@ -331,3 +371,72 @@ func buildEndpointAddressMap(epSubsets []kapi.EndpointSubset) map[epAddressItem]
 
 	return epMap
 }
+
+func configureSvcRouteViaBridge(bridge string) error {
+	gwIPs, _, err := getGatewayNextHops()
+	if err != nil {
+		return fmt.Errorf("unable to get the gateway next hops, error: %v", err)
+	}
+	link, err := util.LinkSetUp(bridge)
+	if err != nil {
+		return fmt.Errorf("unable to get link for %s, error: %v", bridge, err)
+	}
+	for _, subnet := range config.Kubernetes.ServiceCIDRs {
+		gwIP, err := util.MatchIPFamily(utilnet.IsIPv6CIDR(subnet), gwIPs)
+		if err != nil {
+			return fmt.Errorf("unable to find gateway IP for subnet: %v, found IPs: %v", subnet, gwIPs)
+		}
+		err = util.LinkRoutesAdd(link, gwIP[0], []*net.IPNet{subnet})
+		if err != nil && !os.IsExist(err) {
+			return fmt.Errorf("unable to add route for service via shared gw bridge, error: %v", err)
+		}
+	}
+	return nil
+}
+
+func handleUpgrade(quit <-chan struct{}, bridgeName string) {
+	ticker := time.NewTicker(5 * time.Second)
+	klog.Info("Waiting for OVN Masters to upgrade to switchover K8S Service route")
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
you didn't add the global timeout at the end

Reply:
how long should we wait? If there is a big deployment I wonder how long is reasonable.

Reply:
I don't know, 15 mins, or 30 mins, just be conservative but keep it bounded ...

Reply:
also what should we do if we timeout? is it fatal?

Reply:
yeah, exit with error code and a meaningful message ,so the pod/container restarts and the cluster operator/administrator can see the problem and understand what is the issue ...

Reply:
done, thanks!